cmd/snap/cmd_snap_op.go

```diff
@@ -252,10 +252,20 @@ func (mx *channelMixin) setChannelFromCommandline() error {
 		mx.Channel = ch.chName
 	}
 
-	if !strings.Contains(mx.Channel, "/") && mx.Channel != "" && mx.Channel != "edge" && mx.Channel != "beta" && mx.Channel != "candidate" && mx.Channel != "stable" {
-		// shortcut to jump to a different track, e.g.
-		// snap install foo --channel=3.4 # implies 3.4/stable
-		mx.Channel += "/stable"
+	if mx.Channel != "" {
+		ch, err := snap.ParseChannel(mx.Channel, "")
+		if err != nil {
+			if _, ok := err.(snap.DeprecatedChannelStringError); !ok {
+				return err
+			}
+			// TODO: get escapes in here so we can bold the Warning
+			head := i18n.G("Warning:")
+			warn := fill(fmt.Sprintf(i18n.G("Specifying a channel %q is relying on undefined behaviour. Interpreting it as %q for now, but this will change.\n"), mx.Channel, ch.Full()), utf8.RuneCountInString(head)+1) // +1 for the space
```

Comment:
along this path ch.Name and ch.Full() are the same, right?

Reply:
no, this'll print `latest/stable` instead of `stable` for `/stable`; I'm fixing that now.

Reply:
no, given `"/stable"`, `Name` is `"stable"` and `Full()` is `"latest/stable"`. After discussing (and misunderstanding, and realising and fixing it) on IRC, we want `Full()` everywhere for this case.